### ISSUE TITLE
Difference in handling parsing between xml and json

### ISSUE THREAD

**Issue by CorinaCiocanea** [ewoutkramer/fhir-net-api#238]
Hello, 
For the below (invalid)  xml string, the xml parser is able to read the resource while the json parser is not. (The FhirSerializer is also able to write the resource into a json string).

```CSharp
string xml = @"<Patient xmlns = ""http://hl7.org/fhir"" >
                                <name>
                                <family value =""jonas delete3"" />
                                <given value =""test"" />
                                </name>
                                <telecom>
                                    <value value =""jonas3@test.no"" />
                                </telecom>
                                <telecom/>
                        </Patient>";
Patient patient = (Patient)FhirParser.ParseFromXml(xml); //this works fine 
string json = FhirSerializer.SerializeToJson(patient); //this works fine
patient = (Patient)FhirParser.ParseFromJson(json); //this fails
```

**Comment by ewoutkramer** [comment]
Yes, the xml and json parser are totally different internally, since the formats are different, so this may result in different behaviour.  I am not happy about that, but I think both should generally speaking accept as much as possible, so the json should accept a tag like  "name" : {}.  But since json is actually really hard to parse (because of extensions on FHIR primitives in json), I think there is different behaviour here......   

Is this a problem for you?  It's probably costly to fix for me.....
